tutorials/poromechanics.ipynb

```diff
@@ -59,29 +62,38 @@
     "import porepy as pp\n",
     "import numpy as np\n",
     "\n",
-    "class BoundaryConditionsMomentumBalance:\n",
-    "    def bc_type_mechanics(self, sd: pp.Grid) -> pp.BoundaryConditionVectorial:\n",
-    "        \"\"\"Dirichlet boundary condition everywhere.\"\"\"\n",
-    "        # Define boundary faces.\n",
-    "        boundary_faces = self.domain_boundary_sides(sd).all_bf\n",
-    "        bc = pp.BoundaryConditionVectorial(sd, boundary_faces, \"dir\")\n",
-    "        # Default internal BC is Neumann. We change to Dirichlet for the contact\n",
-    "        # problem. I.e., the mortar variable represents the displacement on the fracture\n",
-    "        # faces.\n",
-    "        bc.internal_to_dirichlet(sd)\n",
-    "        return bc\n",
     "\n",
-    "    def bc_values_mechanics(self, subdomains: list[pp.Grid]) -> pp.ad.DenseArray:\n",
-    "        \"\"\"Setting the nonzero Dirichlet values on the north bound.\"\"\"\n",
-    "        # s = super().bc_values_mechanics(subdomains)\n",
-    "        values = []\n",
+    "class BodyForceMixin:\n",
+    "    nd: int\n",
+    "    \"\"\"Ambient dimension.\"\"\"\n",
+    "\n",
+    "    units: pp.Units\n",
+    "\n",
+    "    solid: pp.SolidConstants\n",
+    "\n",
+    "    def body_force(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:\n",
+    "        units = self.units\n",
+    "        vals = []\n",
     "        for sd in subdomains:\n",
-    "            bc = np.zeros((sd.num_faces, self.nd))\n",
-    "            if sd.dim == self.nd:\n",
-    "                bounds = self.domain_boundary_sides(sd)\n",
-    "                bc[bounds.north, 1] = self.solid.convert_units(-0.2, \"m\")\n",
-    "            values.append(bc.ravel())\n",
-    "        return pp.wrap_as_ad_array(np.concatenate(values), \"bc_vals_mechanics\")"
+    "            data = np.zeros((sd.num_cells, self.nd))\n",
+    "\n",
+    "            # We add the source only to the 2D domain and not the fracture.\n",
+    "            if sd.dim == 2:\n",
+    "                # Selecting central cells\n",
+    "                cell_centers = sd.cell_centers\n",
+    "                indices = (\n",
+    "                    (cell_centers[0] > (0.3 / units.m))\n",
+    "                    & (cell_centers[0] < (0.7 / units.m))\n",
+    "                    & (cell_centers[1] > (0.3 / units.m))\n",
+    "                    & (cell_centers[1] < (0.7 / units.m))\n",
+    "                )\n",
+    "\n",
+    "                acceleration = -9.8 * units.m / units.s**2\n",
```

Comment:
```suggestion
    "                acceleration = self.solid.convert_units(-9.8, "m * s^-2")\n",
```
Note that your solution is wrong, as the scaling needs to be inverted (9.8/units.m*units.s**2).

Reply:
What happened here?

Reply:
I accepted your change and then added \n so now it works.